resource/resource.go

```diff
@@ -37,7 +37,7 @@ const (
 
 var (
 	reservedChars     = [...]string{":", "+"} // colons are delimiters for remote names, plus signs are used for WebRTC track names.
-	resRegexValidator = regexp.MustCompile(`^([\w-]+:[\w-]+:(?:[\w-]+))\/?([\w-]+:(?:[\w-]+:)*)?(.+)?$`)
+	resRegexValidator = regexp.MustCompile(`^([\w-]+:[\w-]+:(?:[\w-]+))\/([\w-]+:(?:[\w-]+:)*)?(.+)?$`)
```

Comment:
This `?` indicated that the `/` in `rdk:component:encoder/e` was optional for a fully qualified resource name. It is not optional. See this [regex101 link](https://regex101.com/r/UMXVFW/1) for a better demonstration of how removing `?` avoids matching on `robot2:robot3:robot4:e`.

Reply:
:detective: good catch